index.js

```diff
@@ -181,10 +181,8 @@ module.exports.transform = ({ data, getPluginContext, options }) => {
             return result;
         }
 
-        return {
-            ...result,
-            [canonicalId]: entries[entryId]
-        };
+        result[canonicalId] = entries[entryId];
+        return result;
```

Comment:
this improves 800ms??? 😳

Reply:
What about immutability and functional programming. Can it break something in sourcebit? Or prevent us from doing future performance optimization like comparing if the transform result is different from the original object and then notify of changes, and if not then not notify? By mutating objects it might break it.

Reply:
> this improves 800ms??? 😳

This one is 200-300ms when handling many objects.
The debounce above is 500ms of course.

> What about immutability and functional programming

We're creating `result` as a new object as part of the `reduce`, its scope is limited to the `reduce`.

Reply:
right